packages/fxa-payments-server/src/components/PaymentForm/index.tsx

```diff
@@ -109,21 +70,18 @@ export const PaymentForm = ({
     middleware: validatorMiddlewareReducer,
   });
 
-  const { matchMedia } = useContext(AppContext);
-
   useEffect(() => {
     onMounted(plan);
   }, [onMounted, plan]);
 
-  const engaged = useRef(false);
+  const onChangeEngaged = useCallbackOnce(() => {
```

Comment:
Third refactor: I figured that this call-once callback state tracking might be good to extract into a reusable `useCallbackOnce` hook.